polkadot/node/core/prospective-parachains/src/lib.rs

```diff
@@ -883,18 +882,72 @@ async fn fetch_backing_state<Context>(
 	ctx: &mut Context,
 	relay_parent: Hash,
 	para_id: ParaId,
-) -> JfyiErrorResult<Option<(Constraints, Vec<CandidatePendingAvailability>)>> {
+) -> JfyiErrorResult<Option<(Constraints, Vec<CommittedCandidateReceipt>)>> {
 	let (tx, rx) = oneshot::channel();
 	ctx.send_message(RuntimeApiMessage::Request(
 		relay_parent,
 		RuntimeApiRequest::ParaBackingState(para_id, tx),
 	))
 	.await;
 
-	Ok(rx
+	Ok(rx.await.map_err(JfyiError::RuntimeApiRequestCanceled)??.map(|s| {
+		(
+			From::from(s.constraints),
+			s.pending_availability
+				.into_iter()
+				.map(|c| CommittedCandidateReceipt {
+					descriptor: c.descriptor,
+					commitments: c.commitments,
+				})
+				.collect(),
+		)
+	}))
+}
+
+#[overseer::contextbounds(ProspectiveParachains, prefix = self::overseer)]
+async fn fetch_constraints_and_candidates<Context>(
+	ctx: &mut Context,
+	relay_parent: Hash,
+	para_id: ParaId,
+) -> JfyiErrorResult<Option<(Constraints, Vec<CommittedCandidateReceipt>)>> {
+	match fetch_constraints_and_candidates_inner(ctx, relay_parent, para_id).await {
+		Ok(None) => fetch_backing_state(ctx, relay_parent, para_id).await,
```

Comment:
if the new API returns `None` it probably means that there is something wrong with the para and the backing_state will return `None` also. So it's probably better to log and give up in this case